cylc/flow/rundb.py

```diff
@@ -318,26 +321,44 @@ class CylcWorkflowDAO:
         ],
     }
 
-    def __init__(self, db_file_name, is_public=False):
+    def __init__(
+        self,
+        db_file_name: Union['Path', str],
+        is_public: bool = False,
+        create_tables: bool = False
+    ):
         """Initialise database access object.
 
+        An instance of this class can also be opened as a context manager
+        which will automatically close the DB connection.
+
         Args:
-            db_file_name (str): Path to the database file.
-            is_public (bool): If True, allow retries, etc.
+            db_file_name: Path to the database file.
+            is_public: If True, allow retries, etc.
```

Comment:
Grammatical peeve - if it does something else we should document it.
```suggestion
    is_public: If True, allow retries.
```

Reply:
I think that (retries) is the main consequence (along with associated different reporting of db write errors).

Reply:
I'll commit the suggestion in anticipation of approving this PR...